homework/list.js

```diff
@@ -2,43 +2,57 @@
 
 (function() {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+	function listObject(){
+		this._listNoteContent = [];
+		this._wrapper = document.querySelector('#note-list-wrapper');
+	}
+  
+  listObject.prototype.start = function() {
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
+    this.fetchList()
+    .then(function (data) {
+    	this.updateList(data);
+    	}.bind(this))
+    .then(this.drawList.bind(this))
+    	.then(function () {
+      this.drawList();
+   }.bind(this))
+   .then(function () {
+      this.preloadFirstNote();
+   }.bind(this));
+    //}.bind(this)
+    
     window.addEventListener('click', function(event) {
```

Comment:
Please use `addEventListener` + `handleEvent` pattern instead of binding a function.